File: aoc-2020/Day_01/day_01.py
#!/usr/bin/env python3

f = open("./Day_01_Input.txt", "r")
lst=[int(line) for line in f.read().split()]

print("-----Part 1------")
for line in lst:
    for l in lst:
        if line+l == 2020:
            print("!!!!!!!!!!!!!!!")
            print("FOUND IT!!!")
            print(line, "+", l, "= 2020")
            print(line, "*", l, "=", line*l)
            print("!!!!!!!!!!!!!!!")
            break

print("-----Part 2------")
for line in lst:
    for l in lst:
        for i in lst:
            if line+l+i == 2020:
                print("**********")
                print("FOUNT IT!!!")
                print(line, "+", l, "+", i, "= 2020")
                print(line, "*", l, "*", i, "= ", line*l*i)
                print("**********")
                break
                

f.close()

File: aoc-2020/Day_01/day_01.rs
use std::io::{BufRead, BufReader};
use std::fs::File;

fn main() {
    let mut f = BufReader::new(File::open("Day_01_Input.txt").expect("Failed to open file!"));

    for line in f.lines() {
        for c in line.expect("Failed to read lines!").chars() {
            if (line.parse::<i32>().unwrap() + c.parse::<i32>().unwrap()) == 2020 {
                println!("FOUND!!!");
                println!("{} + {} = 2020 ", line, c);
                println!("{} * {} = {}", line, c, line*c);
            }
        }
    }
}
